src/Group.php

```diff
@@ -119,48 +119,83 @@ public function cleanDBonPurge()
         Rule::cleanForItemCriteria($this, '_groups_id%');
     }
 
-
     public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
     {
+        /**  @var \DBmysql $DB
+         * @var array $CFG_GLPI
+         */
+        global $DB, $CFG_GLPI;
 
-        if (!$withtemplate && self::canView()) {
-            $nb = 0;
-            switch ($item->getType()) {
-                case 'Group':
-                    /** @var Group $item */
-                    $ong = [];
-                    if ($_SESSION['glpishow_count_on_tabs']) {
-                        $nb = countElementsInTable(
-                            $this->getTable(),
-                            ['groups_id' => $item->getID()]
-                        );
-                    }
-                    $ong[4] = self::createTabEntry(__('Child groups'), $nb);
+        if ($withtemplate || !self::canView()) {
+            return '';
+        }
 
+        $ong = [];
+        switch (get_class($item)) {
+            case Group::class:
+                $nb = 0;
+                if ($_SESSION['glpishow_count_on_tabs']) {
+                    $nb = countElementsInTable(
+                        $this->getTable(),
+                        ['groups_id' => $item->getID()]
+                    );
+                }
+                $ong[4] = self::createTabEntry(__('Child groups'), $nb);
+
+                if ($_SESSION['glpishow_count_on_tabs']) {
+                    if ($item->getField('is_itemgroup')) {
+                        $request = 'SELECT id FROM (';
+                        $subqueries = [];
+                        foreach ($CFG_GLPI['linkgroup_types'] as $itemtype_linked) {
+                            if ($DB->fieldExists($itemtype_linked::getTable(), 'groups_id')) {
+                                $subqueries[] = "SELECT id FROM " . $itemtype_linked::getTable() . " WHERE groups_id = " . $item->getID();
+                            }
+                        }
+                        $request .= implode(' UNION ALL ', $subqueries);
+                        $request .= ') AS combined_results';
```

Comment:
Please use the `QueryUnion` and `QuerySubQuery` classes to do this.

Here is an example of usage of these classes:
https://github.com/glpi-project/glpi/blob/308d15fad7f4fa09936e96048ed1206528f01722/src/CommonITILObject.php#L7404-L7456